### ISSUE TITLE
Unable to re-submit a form after correcting a validation error

### ISSUE THREAD

**Issue by jerome2710** [barrelstrength/craft-sprout-forms#411]
### Description (screenshots, logs, ...)
Due to the `SproutFormsDisableSubmitButton` event listener, the submit button gets disabled to prevent multiple submissions. However, when a front-end validation error occurs, the user is unable to re-submit the form again after correcting these errors.

This event listener is added to the DOM when the form is loaded and there is no option to disable this. I believe this is a major bug and should be fixed soon!

### Steps to reproduce

1. Submit the form with validation errors
2. Correct the errors and try to re-submit the form

### Additional info

- Craft version: 3.3.20.1
- PHP version: 7.2.24
- Database driver & version: MySQL 5.5.64
- Plugins & versions: lots, including Sprout Forms 3.7.1.1

**Comment by BenParizek** [comment]
Thanks @jerome2710 

We'll look into improving this. In the meantime, if you need to correct it for your site before we get a release out, you can always set up global override templates and replace the default javascript with your own.

- [Form Template Overrides](https://sprout.barrelstrengthdesign.com/docs/forms/template-overrides.html)
- [The Disable Form Submit JS](https://github.com/barrelstrength/craft-sprout-forms/blob/v3.8.0/src/templates/_components/formtemplates/accessible/form.twig#L84)

**Comment by BenParizek** [comment]
@jerome2710 Can you clarify how you have your forms setup, how you are submitting them, what field you are seeing this behavior with, and what browser you are seeing this in?

I'm not able to recreate this in several scenarios. In browsers that support HTML5 the HTML5 validation is stopping the form submission before several validation errors occur, so the button never gets disabled. If I force disable the HTML5 validation, when the form submits and redirects back with errors and reloads the page and the javascript is reinitialized around that submit button.

Can you help me understand what I'm missing about your scenario?

**Comment by jerome2710** [comment]
@BenParizek I discussed this with our front-enders, but it seems like a generic issue. Regardless of form setup, fields or browser, this behavior will be the same. 

The problem lies within the [disable-submit-button.js](https://github.com/barrelstrength/craft-sprout-forms/blob/v3/src/web/assets/formtemplates/src/js/disable-submit-button.js). If the front-end (eg. HTML5, Bouncer or Parsley) validation finds an error after clicking the submit button, the form will not be sent en messages are shown. The disabled submit button remains, even after fixing the errors.

We are currently solving this problem by overriding the form.twig and not registering the disableSubmitButtonJsUrl. But it feels hacky overriding the entire file for just this issue, we are not getting any updates on form.twig and other people might experience this issue as well.

**Comment by BenParizek** [comment]
@jerome2710 I've updated the default logic for this in the most recent release (v3.8.8) to watch for any changes in input fields and to make sure the submit button does not have the disabled attribute if changes have been made.

I've also got on my list to revisit the form submit javascript logic as we now have several different things taking place on that action on the front-end. Thanks for reporting this.
